src/main/java/tech/stackable/hbase/OpenPolicyAgentAccessController.java

```diff
@@ -0,0 +1,1228 @@
+  }
+
+  @Override
+  public Message preEndpointInvocation(
+      ObserverContext<RegionCoprocessorEnvironment> ctx,
+      Service service,
+      String methodName,
+      Message request) {
+    LOG.debug("preEndpointInvocation not implemented! {}/{}", methodName, request);
+    return request;
+  }
+
+  @Override
+  public void postEndpointInvocation(
+      ObserverContext<RegionCoprocessorEnvironment> ctx,
+      Service service,
+      String methodName,
+      Message request,
+      Message.Builder responseBuilder) {
+    LOG.debug("postEndpointInvocation not implemented! {}/{}", methodName, request);
+  }
+
+  @Override
+  public void preRequestLock(
+      ObserverContext<MasterCoprocessorEnvironment> ctx,
+      String namespace,
+      TableName tableName,
+      RegionInfo[] regionInfos,
+      String description) {
+    LOG.debug("preRequestLock not implemented! {}/{}", tableName, regionInfos);
+  }
+
+  @Override
+  public void preLockHeartbeat(
+      ObserverContext<MasterCoprocessorEnvironment> ctx, TableName tableName, String description) {
+    LOG.debug("preLockHeartbeat not implemented! {}/{}", tableName, description);
+  }
+
+  @Override
+  public void preSetUserQuota(
+      final ObserverContext<MasterCoprocessorEnvironment> ctx,
+      final String userName,
+      final TableName tableName,
+      final GlobalQuotaSettings quotas) {
+    requirePermission(ctx, "setUserTableQuota", tableName, null, null, Action.ADMIN);
+  }
+
+  @Override
+  public void preSetUserQuota(
+      final ObserverContext<MasterCoprocessorEnvironment> ctx,
+      final String userName,
+      final String namespace,
+      final GlobalQuotaSettings quotas) {
+    requirePermission(ctx, "setUserNamespaceQuota", Action.ADMIN);
+  }
+
+  @Override
+  public void preSetTableQuota(
+      final ObserverContext<MasterCoprocessorEnvironment> ctx,
+      final TableName tableName,
+      final GlobalQuotaSettings quotas) {
+    requirePermission(ctx, "setTableQuota", tableName, null, null, Action.ADMIN);
+  }
+
+  @Override
+  public void preSetNamespaceQuota(
+      final ObserverContext<MasterCoprocessorEnvironment> ctx,
+      final String namespace,
+      final GlobalQuotaSettings quotas) {
+    requirePermission(ctx, "setNamespaceQuota", Action.ADMIN);
+  }
+
+  @Override
+  public void preMergeRegions(
+      final ObserverContext<MasterCoprocessorEnvironment> ctx, final RegionInfo[] regionsToMerge) {
+    requirePermission(ctx, "mergeRegions", regionsToMerge[0].getTable(), null, null, Action.ADMIN);
+  }
+
+  @Override
+  public void preGetUserPermissions(
+      ObserverContext<MasterCoprocessorEnvironment> ctx,
+      String userName,
+      String namespace,
+      TableName tableName,
+      byte[] family,
+      byte[] qualifier) {
+    LOG.debug("preGetUserPermissions not implemented! {}/{}", userName, tableName);
+  }
+
+  public void requirePermission(ObserverContext<?> ctx, String request, Action perm) {
```

Comment:
I know you copied it from upstream

```suggestion
  private void requirePermission(ObserverContext<?> ctx, String request, Action perm) {
```